Added shouldDismissPresentedViewController property to options

tspecht opened the pull request:
I recently had the need to stack multiple modal navigation controller in my app. Therefore I added a shouldDismissPresentedViewController property to the options to enable this functionality in my app.
Maybe someone else finds that functionality useful :)

clayallsopp commented on Routable/Routable.h:
```diff
@@ -71,12 +71,14 @@ typedef void (^RouterOpenCallback)(NSDictionary *params);
  @param defaultParams The default parameters which are passed when opening the URL
  @param isRoot The boolean `shouldOpenAsRootViewController` property is set to
  @param isModal The boolean that sets a modal presentation format
+ @param shouldDismissPresentedViewController The boolean inidicating wheter an already existing presentedViewController should be dismissed before presenting a new one
```
some typos in this line

clayallsopp commented on Routable/Routable.m:
```diff
@@ -83,13 +83,16 @@ + (instancetype)routerOptionsWithPresentationStyle: (UIModalPresentationStyle)pr
                                             isRoot: (BOOL)isRoot
-                                           isModal: (BOOL)isModal {
+                                           isModal: (BOOL)isModal
+              shouldDismissPresentedViewController:(BOOL)shouldDismissPresentedViewController
```
nit alignment (the parenthesis)

clayallsopp commented on Routable/Routable.h:
```diff
@@ -71,12 +71,14 @@ typedef void (^RouterOpenCallback)(NSDictionary *params);
                                             isRoot: (BOOL)isRoot
-                                           isModal: (BOOL)isModal;
+                                           isModal: (BOOL)isModal
+              shouldDismissPresentedViewController: (BOOL)shouldDismissPresentedViewController;
```
can we make another method, keeping the original signature (without the `shouldDismissPresentedViewController` arg), and have it propagate the default `YES` value to this new constructor?

just for backwards-compat

clayallsopp commented on Routable/Routable.m:
```diff
@@ -268,15 +292,21 @@ - (void)open:(NSString *)url animated:(BOOL)animated {
-                                            completion:nil];
+        UIViewController* controllerToPresentOn = self.navigationController;
+        
+        if (self.navigationController.presentedViewController) {
```
should we use an additional option for this new behavior? like `options.shouldPresentOnPresentedViewController`

tspecht replied:
this check is actually just present to avoid not showing the new controller at all. The problem is that iOS doesn't allow to push an a controller which is already presenting a controller. So if at this point the navigationController still has a presentedViewController, shouldDismissPresentedViewController must have been false so we really want to push the new controller on top of the presentedViewController

clayallsopp replied:
yup that can happen, but "shouldDismissPresentedViewController must have been false..." makes an assumption that everyone is using Routable for all their modal controllers which may not be true

I'd rather opt-in to this behavior, and if the user hasn't opt'd in and encounters this case (`navigationController.presentedViewController != nil`) then log a message but keep the old behavior (where nothing happens)

clayallsopp commented:
I think this would be useful! if we keep backwards compatibility. I am wary of adding a bunch of options, but I think it's okay for now

can we add a test for this?

tspecht commented:
@clayallsopp thanks for taking a look so fast, I just fixed the typos, added the method back in for backwards compatibility and fixed the indentation. Please also see my other comment.

clayallsopp commented:
awesome! can you add tests? the existing ones should point you in the right direction
